scene/gui/text_edit.cpp

```diff
@@ -1561,6 +1590,10 @@ void TextEdit::gui_input(const Ref<InputEvent> &p_gui_input) {
 					last_dblclk_pos = mb->get_position();
 				}
 
+				if (is_editable() && has_selection() && _is_caret_inside_selection()) {
+					selection.drag_attempt = true;
+				}
```

Comment:
This is needed when you do a double or triple click and you keep the left mouse button pressed on your last click to start dragging.
The code here is really tricky, but I've added the drag and drop feature trying to avoid dangerous refactoring of the actual code.

Reply:
> This is needed when you do a double or triple click and you keep the left mouse button pressed on your last click to start dragging.

We should't start dragging in this use case, it should continue to select text in the given mode.

Reply:
This behavior has been kept from original LineEdit before my changes, this is the discussion https://github.com/godotengine/godot/pull/54339#pullrequestreview-793772482
I can change it in both LineEdit and TextEdit (and perhaps also in my commit for RichTextLabel), I've implemented it only because of KoBeWi comment.
Let me know what should be the expected behavior in these cases in Godot.

Reply:
Looking around most editors have text dragging as a separate action, so I would expect Godot to do the same.

Reply:
I only hope that users do not complain about that change in the previous behavior (I'm referring to LineEdit)

Reply:
I've removed the drag start in case of double or triple click from TextEdit and LineEdit